kernel/src/process.rs

```diff
@@ -34,66 +79,107 @@ use core::cmp::max;
 pub fn load_processes<C: Chip>(
     kernel: &'static Kernel,
     chip: &'static C,
-    start_of_flash: *const u8,
+    app_flash: &'static [u8],
     app_memory: &mut [u8],
     procs: &'static mut [Option<&'static dyn ProcessType>],
     fault_response: FaultResponse,
     _capability: &dyn ProcessManagementCapability,
-) {
-    let mut apps_in_flash_ptr = start_of_flash;
+) -> Result<(), ProcessLoadError> {
+    let mut remaining_flash = app_flash;
     let mut app_memory_ptr = app_memory.as_mut_ptr();
     let mut app_memory_size = app_memory.len();
 
     if config::CONFIG.debug_load_processes {
         debug!(
             "Loading processes from flash={:#010X} into sram=[{:#010X}:{:#010X}]",
-            start_of_flash as usize,
+            app_flash.as_ptr() as usize,
             app_memory_ptr as usize,
             app_memory_ptr as usize + app_memory_size
         );
     }
 
     for i in 0..procs.len() {
         unsafe {
```

Comment:
Did you intend to shrink the size of the `unsafe` block that currently starts just after `for i in 0..procs.len()`?

Reply:
I do, but that would come in a new PR that changes memory to be a slice instead of a pointer, like flash is now.

Reply:
Got it, thanks.